src/main/scala/com/gu/support/workers/encoding/StateCodecs.scala

```diff
@@ -13,8 +14,15 @@ object StateCodecs {
 
   implicit val encodeStatus: Encoder[Status] = Encoder.encodeString.contramap[Status](_.asString)
 
-  implicit val decodeStatus: Decoder[Status] =
-    Decoder.decodeString.emap { identifier => Status.fromString(identifier).toRight(s"Unrecognised status '$identifier'") }
+  implicit val encodeFailureReason: Encoder[CheckoutFailureReason] = Encoder.encodeString.contramap[CheckoutFailureReason](_.asString)
```

Comment:
I'd move `decodeStatus` after `encodeStatus` and `decodeFailureReason` after `encodeFailureReason`